guides/authentication.md

```diff
@@ -212,7 +219,113 @@ client.loginWithOAuth({
 });
 ```
 
-## More
+### More
 
 Here is an article describing more in detail about the authentication process:
 https://doc.cognitedata.com/dev/guides/iam/external-application.html#tokens
+
+## Azure AD auth flow
+
+Azure AD auth flow can be used via the same `loginWithOAuth` method of `CogniteClient`,
+but different `options` structure has to be passed. You'll find what options structure has to be provided in examples below.
+Besides, here list of pre-requirements, that need to be in place to use Azure AD authentication flow:
+
+ - CDF project, which you're trying to log in, has to be configured to accept tokens issued by Azure AD (to be related to the cluster, that accepts Azure AD tokens, `bluefield` for instance)
+ - You should have the `clientId` of your Azure AD application token will be issued for
+ - Cluster your CDF project related to (`bluefield` for instance)
+
+Azure AD authentication flow supports two login types:
+
+ - [With redirect](#authentication-via-redirect)
+ - [With popup](#authentication-via-popup)
+
+### Authentication via redirect
+
+When you authenticate using redirect, auth flow remains quite similar that we have in CDF auth flow.
+But the IdP in this is Azure where user has to log in to prove identity.
```

Comment:
How about:  The IdP in this example is Azure where the user has to sign in to prove the identity.